apps/prairielearn/src/ee/pages/institutionAdminAdmins/institutionAdminAdmins.ts

```diff
@@ -16,7 +35,80 @@ router.get(
       access_as_administrator: res.locals.access_as_administrator,
     });
 
-    res.send(InstitutionAdminAdmins({ institution, resLocals: res.locals }));
+    const rows = await queryValidatedRows(
+      sql.select_admins,
+      { institution_id: institution.id },
+      InstitutionAdminAdminsRowSchema,
+    );
+
+    res.send(
+      InstitutionAdminAdmins({
+        institution,
+        rows,
+        uidsLimit: MAX_UIDS,
+        resLocals: res.locals,
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const institution = await selectAndAuthzInstitutionAsAdmin({
+      institution_id: req.params.institution_id,
+      user_id: res.locals.authn_user.user_id,
+      access_as_administrator: res.locals.access_as_administrator,
+    });
+
+    if (req.body.__action === 'addAdmins') {
+      const uids = parseUidsString(req.body.uids, MAX_UIDS);
+      const validUids: string[] = [];
+      const invalidUids: string[] = [];
+      await runInTransactionAsync(async () => {
+        for (const uid of uids) {
+          const user = await selectUserByUid(uid);
+
+          // Specifically check that the user is in the institution to prevent
+          // someone from enumerating users in other institutions.
+          if (!user || user.institution_id !== institution.id) {
+            invalidUids.push(uid);
+            continue;
+          }
+
+          await insertInstitutionAdministrator({
+            user_id: user.user_id,
+            institution_id: institution.id,
+            authn_user_id: res.locals.authn_user.user_id,
+          });
```

Comment:
Attempting to add a user who is already an institution admin will fail, causing the entire transaction to roll back. Could we handle this case in a nicer way? In PT we do this: https://github.com/PrairieLearnInc/PrairieTest/blob/a9943467d928229ee1a7c908bcf4c58641c5a394/src/models/institution-administrators.ts#L80

Reply:
Added `ON CONFLICT DO NOTHING` and changed function/query name to `ensureInstitutionAdministrator` in 4f17749.